packages/bpk-component-infinite-scroll/src/BpkInfiniteScroll-test.js

```diff
@@ -0,0 +1,198 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import renderer from 'react-test-renderer';
+import { mount } from 'enzyme';
+import toJson from 'enzyme-to-json';
+import BpkInfiniteScroll from './BpkInfiniteScroll';
+import DEFAULT_REQUEST_ELEMENTS from './defaultRequestElements';
+
+global.IntersectionObserver = class {
+  observe() {} // eslint-disable-line
+  unobserve() {  } // eslint-disable-line
+};
+
+describe('BpkInfiniteScroll', () => {
+  const elementsArray = [];
+
+  for (let i = 0; i < 5; i += 1) {
+    elementsArray.push(`Element ${i}`);
+  }
+
+  const List = props => (
+    <div id="list">
+      {props.elements.map(element => <div key={element}>{element}</div>)}
+    </div>
+  );
+
+  List.propTypes = {
+    elements: PropTypes.arrayOf(PropTypes.string).isRequired,
+  };
+
+  const InfiniteList = BpkInfiniteScroll(List);
+
+  it('should render empty correctly', () => {
```

Comment:
The first render is done after the creation of the Component and as the constructor of an object has to be synchronous we don't have elements in the list (because of the onRequestElements returning a Promise).

Reply:
done